include/cnl/_impl/duplex_integer/multiply.h

```diff
@@ -116,12 +115,11 @@ namespace cnl {
         {
             using common_result_type =
                     decltype(_impl::long_multiply<Upper>{}(lhs_upper, rhs_upper));
-            return (_impl::long_multiply<Upper>{}(lhs_upper, rhs_upper) << digits<Upper>::value)
-                 + ((_impl::long_multiply<Upper>{}(lhs_upper, rhs_lower)
-                     + _impl::long_multiply<Upper>{}(lhs_lower, rhs_upper))
-                    << digits<Lower>::value)
-                 + static_cast<common_result_type>(
-                           _impl::long_multiply<Lower>{}(lhs_lower, rhs_lower));
+            auto const ll{static_cast<common_result_type>(_impl::long_multiply<Lower>{}(lhs_lower, rhs_lower))};
```

Comment:
Clang-Format was going to render this expression unreadable if I didn't reduce it. This is a desired task as part of modernisation of the code post-C++11.